kubernetes-deployment/src/chart/templates/_helpers.tpl

```diff
@@ -3,7 +3,7 @@ Since this is a template chart, the chart name will be filled in by the user and
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 */}}
 {{- define "application.fullname" -}}
-{{- .Chart.Name | trunc 63 | trimSuffix "-" }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
```

Comment:
Need to figure out naming. Right now I'm using the "release" name, which in our case ends up being `md_metadata.name_prefix`.  It can be a bit verbose, but it guarantees uniqueness, and I think our policy should be "you shouldn't care what its named if it works".  This also allows users to run 2 of the same applications in the same namespace without there being issues (migrations, blue/green, etc)

```
$ kc get all
NAME                                        READY   STATUS    RESTARTS   AGE
pod/chris-local-test-0000-6d8bfdb49-lxfdv   1/1     Running   0          34m

NAME                            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/chris-local-test-0000   ClusterIP   172.20.135.2   <none>        80/TCP    34m
service/kubernetes              ClusterIP   172.20.0.1     <none>        443/TCP   9h

NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/chris-local-test-0000   1/1     1            1           34m

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/chris-local-test-0000-6d8bfdb49   1         1         1       34m

NAME                                                        REFERENCE                          TARGETS   MINPODS   MAXPODS   REPLICAS   AGE
horizontalpodautoscaler.autoscaling/chris-local-test-0000   Deployment/chris-local-test-0000   0%/80%    1         10        1          34m
```

Reply:
This should definitely be our name. If someone’s biggest problem is typing the name into Kubectl they can enable auto complete

Reply:
It’s also their chart so they can rename it but their mileage will vary if they deploy a similar bundle